src/checkers/inference/solver/util/Constants.java

```diff
@@ -0,0 +1,26 @@
+package checkers.inference.solver.util;
+
+/**
+ * String constants.
+ * 
+ * @author jianchu
+ *
+ */
+public class Constants {
```

Comment:
I prefer to separate them. Some reasons:

1. better type safety: different concepts are guaranteed not mixed-used, e.g a `ConfigConstant` cannot passed in a place required `BackEndTypeEnum`.

2. better customization on each enum class: e.g. I can have some customize methods in a specific enum class, e.g. in `BackEndTypeEnum` I can have a field named `fullyQualifiedName` that manage the fully qualified name of each BackEndType, and have a field `simpleName` that manage the simple name user typed in. Then I can have a static method to validate if a passing simple name is valid to map to an existing BackEndType value. These methods are specific to BackEndType, and is better to not share with other concepts.